challenger_test.go

```diff
@@ -0,0 +1,160 @@
+package aperture
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest/wait"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+var (
+	defaultTimeout = 1 * time.Second
+)
+
+type invoiceStreamMock struct {
+	lnrpc.Lightning_SubscribeInvoicesClient
+
+	updateChan chan *lnrpc.Invoice
+	quit       chan struct{}
+}
+
+func (i *invoiceStreamMock) Recv() (*lnrpc.Invoice, error) {
+	select {
+	case msg := <-i.updateChan:
+		return msg, nil
+
+	case <-i.quit:
+		return nil, context.Canceled
+	}
+}
+
+type mockInvoiceClient struct {
+	invoices   []*lnrpc.Invoice
+	updateChan chan *lnrpc.Invoice
+	quit       chan struct{}
+
+	lastAddIndex uint64
+}
+
+// ListInvoices returns a paginated list of all invoices known to lnd.
+func (m *mockInvoiceClient) ListInvoices(_ context.Context,
+	_ *lnrpc.ListInvoiceRequest,
+	_ ...grpc.CallOption) (*lnrpc.ListInvoiceResponse, error) {
+
+	return &lnrpc.ListInvoiceResponse{
+		Invoices: m.invoices,
+	}, nil
+}
+
+// SubscribeInvoices subscribes to updates on invoices.
+func (m *mockInvoiceClient) SubscribeInvoices(_ context.Context,
+	in *lnrpc.InvoiceSubscription, _ ...grpc.CallOption) (
+	lnrpc.Lightning_SubscribeInvoicesClient, error) {
+
+	m.lastAddIndex = in.AddIndex
+
+	return &invoiceStreamMock{
+		updateChan: m.updateChan,
+		quit:       m.quit,
+	}, nil
+}
+
+// AddInvoice adds a new invoice to lnd.
+func (m *mockInvoiceClient) AddInvoice(_ context.Context, in *lnrpc.Invoice,
+	_ ...grpc.CallOption) (*lnrpc.AddInvoiceResponse, error) {
+
+	m.invoices = append(m.invoices, in)
+
+	return &lnrpc.AddInvoiceResponse{
+		RHash:          in.RHash,
+		PaymentRequest: in.PaymentRequest,
+		AddIndex:       uint64(len(m.invoices) - 1),
+	}, nil
+}
+
+func (m *mockInvoiceClient) stop() {
+	close(m.quit)
+}
+
+func newChallenger() (*LndChallenger, *mockInvoiceClient) {
+	mockClient := &mockInvoiceClient{
+		updateChan: make(chan *lnrpc.Invoice),
+		quit:       make(chan struct{}),
+	}
+	genInvoiceReq := func(price int64) (*lnrpc.Invoice, error) {
+		return &lnrpc.Invoice{
+			PaymentRequest: "foo",
+			RHash:          lntypes.ZeroHash[:],
+			AddIndex:       99,
+			State:          lnrpc.Invoice_OPEN,
+		}, nil
+	}
+	return &LndChallenger{
+		client:        mockClient,
+		genInvoiceReq: genInvoiceReq,
+		invoiceStates: make(map[lntypes.Hash]lnrpc.Invoice_InvoiceState),
+		quit:          make(chan struct{}),
+	}, mockClient
+}
+
+func TestLndChallenger(t *testing.T) {
```

Comment:
Nit: commit message isn't fully descriptive of the changes, namely the added test.